src/view/treeNodes/pullRequestNode.ts

```diff
@@ -504,265 +573,206 @@ export class PRNode extends TreeNode {
-
-			let isBase = !!(params && params.isBase);
-			let position = mapHeadLineToDiffHunkPosition(fileChange.diffHunks, '', range.start.line + 1, isBase);
-
-			if (position < 0) {
-				throw new Error('Comment position cannot be negative');
+	public async editComment(thread: vscode.CommentThread, comment: vscode.Comment): Promise<void> {
+		if (await this._prManager.authenticate() && this._commentController!.inputBox) {
+			const existingComment = (comment as (vscode.Comment & { _rawComment: Comment }))._rawComment;
+			if (!existingComment) {
+				throw new Error('Unable to find comment');
 			}
 
-			// there is no thread Id, which means it's a new thread
-			const rawComment = await this._prManager.createComment(this.pullRequestModel, text, params!.fileName, position);
-			const comment = convertToVSCodeComment(rawComment!);
-
-			fileChange.comments.push(rawComment!);
-
-			let commentThread: vscode.CommentThread = {
-				threadId: comment.commentId,
-				resource: uri,
-				range: range,
-				comments: [comment]
-			};
+			const rawComment = await this._prManager.editReviewComment(this.pullRequestModel, existingComment, this._commentController!.inputBox!.value);
+			const vscodeComment = convertToVSCodeComment(rawComment, undefined);
 
-			return commentThread;
-		} catch (e) {
-			throw new Error(formatError(e));
-		}
-	}
-
-	private async editComment(document: vscode.TextDocument, comment: vscode.Comment, text: string): Promise<void> {
-		const fileChange = this.findMatchingFileNode(document.uri);
-		const existingComment = fileChange.comments.find(c => c.id.toString() === comment.commentId);
-		if (!existingComment) {
-			throw new Error('Unable to find comment');
-		}
-
-		const rawComment = await this._prManager.editReviewComment(this.pullRequestModel, existingComment, text);
+			let newComments = thread.comments.map(cmt => {
+				if (cmt.commentId === vscodeComment.commentId) {
+					vscodeComment.editCommand = getEditCommand(thread, vscodeComment, this);
+					vscodeComment.deleteCommand = getDeleteCommand(thread, vscodeComment, this);
+					return vscodeComment;
+				}
 
-		const index = fileChange.comments.findIndex(c => c.id.toString() === comment.commentId);
-		if (index > -1) {
-			fileChange.comments.splice(index, 1, rawComment);
+				return cmt;
+			});
+			thread.comments = newComments;
+			updateCommentThreadLabel(thread);
 		}
 	}
 
-	private async deleteComment(document: vscode.TextDocument, comment: vscode.Comment): Promise<void> {
-		const fileChange = this.findMatchingFileNode(document.uri);
-
+	public async deleteComment(thread: vscode.CommentThread, comment: vscode.Comment): Promise<void> {
 		await this._prManager.deleteReviewComment(this.pullRequestModel, comment.commentId);
-		const index = fileChange.comments.findIndex(c => c.id.toString() === comment.commentId);
+		const index = thread.comments.findIndex(c => c.commentId === comment.commentId);
 		if (index > -1) {
```

Comment:
If the comment is the last in the thread, we should also dispose the thread. Ditto in the reviewDocumentCommentProvider